services/hackernews/hackernews-user-karma.service.js

```diff
@@ -0,0 +1,70 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { BaseJsonService, NotFound } from '../index.js'
+import { anyInteger } from '../validators.js'
+
+const schema = Joi.object({
+  id: Joi.string(),
+  karma: anyInteger,
+})
+  .allow(null)
+  .required()
+
+export default class HackerNewsUserKarma extends BaseJsonService {
+  static category = 'social'
+
+  static route = {
+    base: 'hackernews/user-karma',
+    pattern: ':userid',
+  }
+
+  static examples = [
+    {
+      title: 'HackerNews User Karma',
+      namedParams: { userid: 'pg' },
+      staticPreview: this.render({ userid: 'pg', karma: '156359' }),
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'HackerNews User Karma',
+    namedLogo: 'ycombinator',
+  }
+
+  static render({ karma, userid }) {
+    return {
+      label: userid,
+      message: metric(karma),
+      color: 'red',
```

Comment:
Why always red here? This gets used if you request the non-social version. Maybe `brightgreen` for positive, `red` for negative and `orange` for zero would make sense?